server/modules/taxonomy/repository/taxonomyRepository.js

```diff
@@ -93,22 +96,33 @@ export const countTaxaByTaxonomyUuid = async (surveyId, taxonomyUuid, draft = fa
 
 export const fetchTaxaWithVernacularNames = async (surveyId, taxonomyUuid, draft = false, limit = null, offset = 0, client = db) =>
   await client.map(`
+      WITH vernacular_names AS (
+        SELECT
+          vn.taxon_uuid,
+          (vn.props || vn.props_draft)->>'${TaxonVernacularName.keysProps.lang}' as lang,
+          array_agg(
+            json_build_object(
+              '${TaxonVernacularName.keys.uuid}', vn.uuid,
+              '${TaxonVernacularName.keys.props}', vn.props || vn.props_draft
+            )
+          ) as names
+        FROM ${getSurveyDBSchema(surveyId)}.taxon_vernacular_name vn
+        GROUP BY vn.taxon_uuid, (vn.props || vn.props_draft)->>'${TaxonVernacularName.keysProps.lang}'
```

Comment:
why are you grouping by lang? don't you return an array e.g. :[{lang:'ita', value:'name1'},{lang:'ita', value:'name2'}]?
or is it easier to use [{'ita':['name1','name2']}]?

Reply:
it's easier to group them by lang